File: app/src/main/java/buggy/plandroid/com/plandroidbugs/room/UserEntity.java
package buggy.plandroid.com.plandroidbugs.room;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Map;

@Entity(tableName = "users")
public class UserEntity {

    @NonNull @PrimaryKey @ColumnInfo(name = "uid") public String userId;
    @ColumnInfo(name = "email") public String email;
    @ColumnInfo(name = "first_name") public String firstName;
    @ColumnInfo(name = "last_name") public String lastName;
    @ColumnInfo(name = "company") public String company;
    @ColumnInfo(name = "title") public String title;
    @ColumnInfo(name = "language") public String language;
    @ColumnInfo(name = "role_uid") public String roleUid;
}

File: app/src/main/java/buggy/plandroid/com/plandroidbugs/room/UserEntity.kt
package buggy.plandroid.com.plandroidbugs.room

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "users")
data class UserEntity(
    @PrimaryKey
    @ColumnInfo(name = "uid")
    val userId: String,

    @ColumnInfo(name = "email")
    val email: String,

    @ColumnInfo(name = "first_name")
    val firstName: String,

    @ColumnInfo(name = "last_name")
    val lastName: String,

    @ColumnInfo(name = "company")
    val company: String,

    @ColumnInfo(name = "title")
    val title: String,

    @ColumnInfo(name = "language")
    val language: String,

    @ColumnInfo(name = "role_uid")
    val roleUid: String,
)
